File: src/main/java/com/natia/secretmod/utils/SkyblockMob.java
package com.natia.secretmod.utils;

import com.natia.secretmod.features.slayers.VoidGloom;
import net.minecraft.client.Minecraft;
import net.minecraft.entity.Entity;
import net.minecraft.entity.item.EntityArmorStand;
import net.minecraft.util.StringUtils;

public class SkyblockMob {

    private String health;
    private Entity entity;
    public SkyblockMob(String mobName, Entity entity) {
        this.entity = entity;
        for (EntityArmorStand e : Minecraft.getMinecraft().theWorld.getEntitiesWithinAABB(EntityArmorStand.class, entity.getEntityBoundingBox().expand(2, 5, 2))) {
            if (StringUtils.stripControlCodes(e.getName()).contains(mobName)) {
                health = e.getName();
                break;
            }
        }
    }

    public Entity getEntity() {
        return entity;
    }

    public String getHealth() {
        return health;
    }
}

File: src/main/kotlin/natia/skytweaks/utils/SkyblockMob.kt
package natia.skytweaks.utils

import natia.skytweaks.SecretUtils
import net.minecraft.client.Minecraft
import net.minecraft.entity.Entity
import net.minecraft.entity.item.EntityArmorStand
import net.minecraft.util.BlockPos
import net.minecraft.util.StringUtils

class SkyblockMob(mobName: String, entity: Class<out Entity>) {

    var health: String? = null
        private set
    var entity: Entity? = null
        private set

    init {
        val entities = Minecraft.getMinecraft().theWorld.getEntities(entity) { e ->
            val playerPos = BlockPos(Minecraft.getMinecraft().thePlayer.posX, Minecraft.getMinecraft().thePlayer.posY, Minecraft.getMinecraft().thePlayer.posZ)
            assert(e != null)
            SecretUtils.withinRange(ENTITYRANGE, e!!.position, playerPos)
        }
        for (entity1 in entities) {
            for (e in Minecraft.getMinecraft().theWorld.getEntitiesWithinAABB(EntityArmorStand::class.java, entity1.entityBoundingBox.expand(15.0, 10.0, 15.0))) {
                if (StringUtils.stripControlCodes(e.name).contains(mobName)) {
                    health = e.name
                    this.entity = entity1
                    break
                }
            }
        }
    }

    fun foundMob(): Boolean {
        return health != null && entity != null
    }

    companion object {

        var ENTITYRANGE = BlockPos(30, 15, 30)
    }
}
